javalib/src/main/scala/java/util/Objects.scala

```diff
@@ -103,9 +107,78 @@ object Objects {
   def checkFromIndexSize(fromIndex: Long, size: Long, length: Long): Long = {
     if ((length | fromIndex | size) < 0L || size > length - fromIndex) {
       throw new IndexOutOfBoundsException(
-        s"Range [$fromIndex, $fromIndex + $size] out of bounds for length $length"
+        s"Range [$fromIndex, $fromIndex + $size) out of bounds for length $length"
       )
     }
     fromIndex
   }
+
+  /** since JDK9 */
+  def checkFromToIndex(fromIndex: Int, toIndex: Int, length: Int): Int = {
+    if ((fromIndex < 0) || (fromIndex > toIndex) || (toIndex > length)) {
+      throw new IndexOutOfBoundsException(
+        s"Range [$fromIndex, $toIndex) out of bounds for length $length"
+      )
+    }
+
+    fromIndex
+  }
+
+  /** since JDK16 */
+  def checkFromToIndex(fromIndex: Long, toIndex: Long, length: Long): Long = {
+    if ((fromIndex < 0L) || (fromIndex > toIndex) || (toIndex > length)) {
+      throw new IndexOutOfBoundsException(
+        s"Range [$fromIndex, $toIndex) out of bounds for length $length"
+      )
+    }
+
+    fromIndex
+  }
+
+  /** since JDK9 */
+  def checkIndex(index: Int, length: Int): Int = {
+    if ((index < 0) || (index >= length)) {
+      throw new IndexOutOfBoundsException(
+        s"Index $index out of bounds for length $length"
+      )
+    }
+
+    index
+  }
+
+  /** since JDK16 */
+  def checkIndex(index: Long, length: Long): Long = {
+    if ((index < 0L) || (index >= length)) {
+      throw new IndexOutOfBoundsException(
+        s"Index $index out of bounds for length $length"
+      )
+    }
+
+    index
+  }
+
+  /** since JDK9 */
+  def requireNonNullElse[T](obj: T, defaultObj: T): T = {
+    if (obj != null) obj
+    else Objects.requireNonNull[T](defaultObj, "defaultObj")
+  }
+
+  /** since JDK9 */
+  def requireNonNullElseGet[T](obj: T, supplier: Supplier[_ <: T]): T = {
+    if (obj != null) obj
+    else {
+      Objects.requireNonNull(supplier, "supplier")
+      Objects.requireNonNull(supplier.get(), "supplier.get()")
+    }
+  }
+
+  /** since JDK19 */
```

Comment:
I think these are suppose to be `@since JDK19` so it shows up correctly. I haven't tested but that is how it was in Javadoc at least and I think Scaladoc inherited that.

Reply:
Right you are, thank you for catching that.

New commit in progress. I'll check build log tomorrow.